File: src/Learnify/Learnify.Core/Dto/Course/LessonDtos/LessonUpdateResponse.cs
using Learnify.Core.Dto.Attachment;
using Learnify.Core.Dto.Course.QuizQuestion;
using Learnify.Core.Dto.Course.Video;
using Learnify.Core.Enums;

namespace Learnify.Core.Dto.Course.LessonDtos;

/// <summary>
/// LessonUpdateResponse
/// </summary>
public class LessonUpdateResponse
{
    public string Id { get; set; }
    public int ParagraphId { get; set; }
    public string Title { get; set; }
    public string EditedLessonId { get; set; }
    public string OriginalLessonId { get; set; }
    public bool IsDraft { get; set; }
    public Language PrimaryLanguage { get; set; }

    public VideoResponse Video { get; set; }
    
    public IEnumerable<QuizQuestionUpdateResponse> Quizzes { get; set; }
}

File: front/client/src/app/Models/Course/Lesson/LessonUpdateResponse.ts
import { Language } from "../../enums/Language";
import { QuizQuestionResponse } from "./QuizQuestion/QuizQuestionResponse";
import { QuizQuestionUpdateResponse } from "./QuizQuestion/QuizQuestionUpdateResponse";
import { VideoResponse } from "./Video/VideoResponse";

export interface LessonUpdateResponse {
    id: string | null;
    paragraphId: number;
    title: string;
    editedLessonId: string;
    originalLessonId: string;
    isDraft: boolean;
    primaryLanguage: Language;
    video: VideoResponse;
    quizzes: QuizQuestionUpdateResponse[];
}